staging/src/k8s.io/pod-security-admission/test/fixtures_hostPorts.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/pod-security-admission/api"
+)
+
+/*
+TODO: include field paths in reflect-based unit test
+
+containerFields: []string{
+	`securityContext.capabilities.add`,
+},
+*/
+
+// ensureHostPorts ensures the pod and all initContainers and containers have a non-nil Ports.
+func ensureHostPorts(p *corev1.Pod) *corev1.Pod {
```

Comment:
I don't think this function is necessary... empty list and nil are not distinguished from each other in API requests or validation; we can assign directly to the ports field of the test containers without needing to check nil first

Reply:
Oh I did not know that. Removed it now